dataflow/src/org/checkerframework/dataflow/analysis/Analysis.java

```diff
@@ -422,15 +426,18 @@ protected boolean updateNodeValues(Node node, TransferResult<A, S> transferResul
 
     /** Initialize the analysis with a new control flow graph. */
     protected void init(ControlFlowGraph cfg) {
+        thenStores.clear();
+        elseStores.clear();
+        if (blockCount != null) {
+            blockCount.clear();
+        }
+        inputs.clear();
+        storesAtReturnStatements.clear();
+        //        nodeValues.clear();
```

Comment:
This is what "fixed" the crash.   Here's why:

The Value Checker runs dataflow on each block in the current class tree, clearing out the nodeValues before each block.  After the Value Checker checks the current class tree, the Upper Bound Checker requests a type from the Value Checker.  When the Value Checker re-runs dataflow to get that type, the nodeValues for that block have already been cleared.

I don't think this is the right "fix", but maybe with this information you can decide how to proceed.

Reply:
Thanks for figuring this out. I agree that this isn't the right fix :-) All results that need to survive should already have been added to `flowResults` and then should be available for other analyses.

Could you point me to where the Upper Bound Checker interacts with the Value Checker?

Reply:
Here's one example:
UpperBoundVisitor.relaxedCommonAssignmentCheck(UpperBoundVisitor.java:262)

But, the Index Checker is full of such interactions.